packages/ifpack2/src/Ifpack2_AdditiveSchwarz_def.hpp

```diff
@@ -425,6 +425,19 @@ apply (const Tpetra::MultiVector<scalar_type,local_ordinal_type,global_ordinal_t
     if (C_.get () == nullptr || C_->getNumVectors () != numVectors) {
       C_.reset (new MV (Y.getMap (), numVectors, false));
     }
+    // If taking averages in overlap region, we need to compute
+    // the number of procs who have a copy of each overlap dof
+    Teuchos::ArrayRCP<double>  dataNumOverlapCopies;
+    if (IsOverlapping_ && AvgOverlap_) {
+      if (num_overlap_copies_.get()  == nullptr) {
+        num_overlap_copies_.reset (new MV (Y.getMap (), 1, false));
+        RCP<MV> onesVec( new MV(OverlappingMatrix_->getRowMap(), 1, false) );
+        onesVec->putScalar(Teuchos::ScalarTraits<scalar_type>::one());
+        rcp_dynamic_cast<OverlappingRowMatrix<row_matrix_type>> (OverlappingMatrix_)->exportMultiVector (*onesVec, *(num_overlap_copies_.get ()), CombineMode_);
+      }
+      dataNumOverlapCopies = num_overlap_copies_.get ()->getDataNonConst(0);
```

Comment:
It's weird that only the Zoltan2Sphynx [build](https://trilinos-cdash.sandia.gov/viewBuildError.php?buildid=396529) fails for SC=complex<double>.

Reply:
This is being tracked in https://github.com/trilinos/Trilinos/issues/10823#issuecomment-1206856350.